lib/net/pool.js

```diff
@@ -1883,6 +1894,142 @@ class Pool extends EventEmitter {
     }
   }
 
+  /**
+   * Handle peer `getcfilters` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {GetCFiltersPacket} packet
+   */
+
+  async handleGetCFilters(peer, packet) {
+    if (this.options.selfish)
+      return;
+
+    if (this.chain.options.spv)
+      return;
+
+    if (this.chain.options.prune)
+      return;
+
+    if (this.chain.options.index)
+      return;
```

Comment:
I think this can all be compressed into a single check. Probably for `this.options.bip157` or something. Look through pool.js for how we handle bloom filter and merkleblock requests I think we just check for `this.options.bip37`. This also means that `this.options.bip157` MUST only be set when we ARE all a full node with the indexer running.

Reply:
...we also should probably ignore these requests if our chain is not in sync yet.